### PR TITLE
[WPB-6073] cleanup haskell pins

### PR DESCRIPTION
**Pull request by MangoIV**
https://wearezeta.atlassian.net/browse/WPB-6073

# libraries that need investigating 

I have tried to sort these roughly by importance

- bloodhound (immense divergence, tests don't pass)
- warp (tests don't pass)
- saml2-web-sso (tests don't pass) 
- amqp (tests don't pass)
- cql-io (tests don't pass) 
- hspec-junit-formatter (tests don't pass) 
- markov-chain-usage-model (tests don't pass) 
- openapi3 (tests don't pass) 
- quickcheck-state-machine (tests don't pass) 
- transitive-anns (tests are flaky) 
- wai-route (it has been noted to get rid of it for a while and we depend on a quite old version) 
- tasty (immense divergence) 
- there's an entire family of libraries that are made by thoralf wittner that we still have in use, commonly as a fork that may also already be years old, it doesn't seem like any of these libraries get any maintenance. Perhaps we can consider taking over maintenance for those

# removed/ updated pins

- amqp: has landed in nixpkgs
- invertible: has landed in nixpkgs
- tls: has landed in nixpkgs
- hoogle/ ghc-source-gen: directly from hackage, has not landed in nixpkgs
- polysemy: newer version on hackage
- hpack: landed in nixpkgs
- hsopenssl: newer version on hackage 
- http2: newer version on hackage 
- network-conduit-tls: landed in nixpkgs
- warp-tls: landed in nixpkgs

# removed overrides 

- kind-generics-th
- http-client
- hsaml2 
- crypton-connection
- transitive-anns
- wai-predicates
- wai-middleware-prometheus
- type-errors
- text-short
- text-icu-translit
- singletons-base
- singletons-th
- servant
- servant-client
- servant-client-core
- servant-foreign
- servant-multipart 
- servant-swagger-ui-core
- servant-swagger-ui 
- polysemy
- polysemy-plugin
- polysemy-check
- monoidal-containers
- invertible
- hashtables
- ghc-source-gen

# remaining pins and their current state

- transitive-anns: we maintain this library by ourselves 
- cryptobox-haskell: we maintain this library by ourselves
- saml2-web-sso: we maintain this library ourselves 
- bloodhound: has diverged wildly, should probably be rebased on upstream and/or merged to it, see [WPB-6168: bloodhound - switch to upstream Todo](https://wearezeta.atlassian.net/browse/WPB-6168) 
- HaskellNet-SSL: PR open, upstream seems abandoned
- hsaml2: actively maintained, should probably  be upstreamed
- hspec-wai: PR open, upstream seems abandoned 
- cql: PR open, upstream seems abandoned, maintainer (thoralf wittner) searches for other maintainers 
- cql-io: PR open, upstream seems abandoned, maintainer (thoralf wittner) searches for other maintainers 
- wai-predicates: missing upstream PR, seems likely abandoned, though
- wai-routing: we use upstream but it appears abandoned as well, mr wittner doesn’t upstream anything to hackage anymore (latest update on hackage 2016, latest commit (the one we use) 2018)
- tasty: [our PR ](https://github.com/UnkindPartition/tasty/pull/351)will not get accepted, we should consolidate also implementing our change to HUnit as requested or think about whether we want to continue maintaining our fork which has diverged a lot
- servant-openapi3: we have a PR open and the project seems to me more or less maintained, there hasn’t been an answer from the maintainers yet, though. Not much we can do here
- postie: PR open, our PR is missing a hackage release
- tinylog: part of the thoralf wittner zoo of libraries, probably abandoned, no PR open to test it, though
- tasty-ant-xml: PR open, the maintainer is occasionally seen, so probably not abandoned. I bumped the PR 
- text-icu-translit: project seems to be abandoned
- warp: PR is somewhat recent (1 month) and project doesn’t seem to be abandoned
- wai-route: note says we should get rid of it, currently only brig and metrics-wai depend on it
- ghc-source-gen, hoogle, safe are not yet in nixpkgs but already released on hackage

# new pins 
- safe, dependency on hoogle which we now pull from hackage instead of from the upstream git repo 

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Comment by fisx** [comment]
nice!

minor note: wai-route and wai-predicates should be replaced by servant at some point (not this PR!)

File: nix/haskell-pins.nix [hunk 1]
```diff
@@ -59,86 +59,108 @@
...
     wai-predicates = {
       src = fetchgit {
-        url = "https://gitlab.com/wireapp/forks/wai-predicates.git";
+        url = "https://gitlab.com/wireapp/forks/wai-predicates";
         rev = "ff95282a982ab45cced70656475eaf2cefaa26ea";
         sha256 = "sha256-x2XSv2+/+DG9FXN8hfUWGNIO7V4iBhlzYz19WWKaLKQ=";
       };
     };
+
+    # we use upstream, but has not been uploaded to hackage since 2016
```
Comment (elland) [inline nix/haskell-pins.nix:155]: Add a comment that this will go away once servantification is complete cc @fisx

File: nix/manual-overrides.nix [hunk 2]
```diff
@@ -1,95 +1,69 @@
...
-  crypton-connection = hlib.markUnbroken hsuper.crypton-connection;
-  # Patched dependency on crypton-connection
-  HaskellNet-SSL = hsuper.HaskellNet-SSL;
   warp = hlib.dontCheck hsuper.warp;
 
-  # PR with fix: https://github.com/freckle/hspec-junit-formatter/pull/23
-  hspec-junit-formatter = hlib.markUnbroken (hlib.dontCheck hsuper.hspec-junit-formatter);
+  # ---------------------
+  # need to be jailbroken
+  # (these need to be fixed upstream eventually)
```
Comment (elland) [inline nix/manual-overrides.nix:28]: `FUTUREWORK`?

File: nix/haskell-pins.nix (hunk 1 above)
Comment (MangoIV) [reply nix/haskell-pins.nix:155]: :+1:

**Review by elland — APPROVED** [review]

**Pull request merged**